ReactNativeClient/lib/services/InteropService_Exporter_Json.js

```diff
@@ -0,0 +1,80 @@
+const InteropService_Exporter_Base = require('lib/services/InteropService_Exporter_Base');
+const { basename, filename } = require('lib/path-utils.js');
+const { shim } = require('lib/shim');
+
+class InteropService_Exporter_Json extends InteropService_Exporter_Base {
+
+	async init(destDir) {
+		this.destDir_ = destDir;
+		this.resourceDir_ = destDir ? destDir + '/resources' : null;
+
+		await shim.fsDriver().mkdir(this.destDir_);
+		await shim.fsDriver().mkdir(this.resourceDir_);
+	}
+
+	async processItem(ItemClass, item) {
+		const obj = await this.buildPlainObjectForJson_(ItemClass, item);
```

Comment:
I would have expected that simply calling JSON.stringfy(item) woudl be enough. Any reason this buildPlainObjectFromJson function is needed? Are there keys in item that should not be serialized? Note that we also need to export `type_` or else it's not possible to know the object type.

Reply:
stringify(item) could work, but I think it's less fragile to have more control over what is written to disk, in case future code ever changes the structure of 'item'. I can simplify the method a bit though by creating a simpler object. Yes, the export does include type_, and it's verified by the test.

Reply:
> stringify(item) could work, but I think it's less fragile to have more control over what is written to disk, in case future code ever changes the structure of 'item'.

I see what you mean and in some cases it would make sense to follow your approach. However in this case, the structure of the items is quite fixed - there might be new properties added but that will need to be serialised too, and on the other hand there won't be anything unusual added to it or any special properties (like type_) so it's safe to simply serialise it with JSON.stringfy and save it to disk. The item is basically exactly a row in the database table, except for the type_ property.